### DISCUSSION TITLE
Losing my mind over `Failed to execute .../on-add.hook Error: no such file or directory`

### DISCUSSION THREAD

**Discussion by akim-13** [GothenburgBitFactory/taskwarrior#3134] [Community support] [answered]
I'm trying to integrate taskwarrior with Android via Termux. I create a notification with `termux-notification` to have a "Create new task" button there, which in turn executes a script that has some taskwarrior commands in it. If I try to run the script by itself everything works fine, however when it is executed from that notification, I get the error described in the title, where it shows the absolute path to the hook and says it doesn't exist. This is relevant, since from `termux-notification --help-action`:

>The action is run in a different environment (not a subshell). Thus your environment is lost (most notable $PATH), and ~/.profile is not sourced either.

Nevertheless, I'm not sure whose problem it is (TW or Termux), since when I try to `cat` the hook in question from inside the same script, it does so successfully, proving that the script in fact exists even within the different `termux-notification` environment. 

I'm not sure how to debug this, so if you have any ideas, help would be greatly appreciated. Please request any info that you think might help to resolve the issue.

Edit:
It also doesn't make sense for TW to say that this absolute path to the script doesn't exist, since as far as I'm aware, it is not specified anywhere else, i.e. if the script actually did not exists, instead of throwing an error TW just wouldn't be able to find and execute it.

**Comment by sdondley** [comment]
Not familiar with termux at all but if you are loing $PATH information, make sure your script has full paths to the commands you want to run.

Instead of `~/bin/command it would be `/full/path/ot/the/bin/command`

**Reply by akim-13** [reply]
Thank you, but that's not it. I manually `export $PATH` at the beginning of the script, and even after doing so, I use absolute paths wherever possible. I also set `chmod 777` to the **hook** (not the script), so it can't be a permission issue either.

What I really don't understand is how this error is possible in the first place. The names of the hooks aren't specified anywhere but in the hooks directory itself, are they? If so, how in the world TW can say that it can't find a specific hook????

**Reply by djmitche** [reply]
I'd guess that the script has a `#!` line at the top that is pointing to something that doesn't exist? I've been confused by that kind of error before. The EEXIST can occur either because the file itself doesn't exist (but `cat` says it does) or because the kernel can't load the executable specified in the `#!` line.  Android's a particularly funny flavor of Linux, so perhaps there's some security mechanism at play here?

**Reply by akim-13** [reply]
This is actually a very interesting point. I do not have a shebang in the beginning because if I do put it there, the script just stops working when run from the "New task" termux-notification button (but does work if run directly as `./script.sh`). 

It's a fun situation to be in, since with the shebang TW works but Termux doesn't and vice versa... I guess there is no way around it on the TW side of things, right?

**Reply by djmitche** [reply]
No, that sounds very much like an Android/termux issue.

**Reply by akim-13** [reply]
Thank you for your help!
Edit: celebrated too early, will elaborate soon.

**Reply by akim-13** [reply, marked as answer]
Solved, just changed the usual shebang inside the hook to absolute path. See the [relevant discussion](https://github.com/termux/termux-app/discussions/3533#discussioncomment-6638207).
